packages/@aws-cdk/aws-glue/lib/code.ts

```diff
@@ -42,7 +42,8 @@ export class S3Code extends Code {
     super();
   }
 
-  public bind(_scope: constructs.Construct): CodeConfig {
+  public bind(job: Job): CodeConfig {
+    this.bucket.grantRead(job);
```

Comment:
I get that this tries to do least-privilege. But, wouldn't the policy get bloated if we think about having several `extra-*` assets?

Additionally, for the `AssetCode`, the `s3assets.Asset` doesn't expose a similar method, so it's at the bucket level.

So, why not keep both granting at the bucket level?

Reply:
I think security > policy bloat. The comment for `aws-s3-assets.Asset` remarks:
>   we give permissions on all files in the bucket since we don't want to accidentally revoke permission on old versions when deploying a new version (for example, when using Lambda traffic shifting).

I don't believe this reasoning applies to this case since there is no consistent "asset bucket" where assets will be published and Glue does not have this concept of traffic shifting

Reply:
In the actual use case I have, we don't use `s3assets.Asset`, instead we upload our different jars to the same bucket - which I think would be more natural to do.
In all cases, the change requested is now in https://github.com/aws/aws-cdk/pull/12506/commits/03286152d6fcbfd216e8bd5917d62e741ea3a8d2 and I hope that in practice this will not be a problem :)